src/helpers/getValidationErrors.js

```diff
@@ -53,32 +48,32 @@ function formatMsg(file, msg) {
 }
 
 function validateDescriptorProvider(file, provider, {
+  actualComponents,
   projectConfig,
   projectRoot,
 }) {
   if (typeof provider !== 'function') {
     throw new TypeError('provider must be a function');
   }
 
-  if (provider.length !== 1 && provider.length !== 2) {
-    throw new RangeError(`provider function must take exactly 1 or 2 arguments: takes ${provider.length}`);
+  if (provider.length !== 0 && provider.length !== 1) {
```

Comment:
Out of curiosity, did you not do `if (provider.length > 1)` so that the check better mirrored the error message?

Reply:
i guess i could; i was more mirroring the previous structure.